test/rules/src/https_everywhere_checker/check_rules.py

```diff
@@ -394,7 +399,24 @@ def cli():
 				logging.error(problem)
 		trie.addRuleset(ruleset)
 		rulesets.append(ruleset)
-	
+
+	if checkConflictingRules:
+		logging.debug("Checking ruleset conflicts")
+		for ruleset in rulesets:
+			for test in ruleset.tests:
+				matches = list(trie.transformUrlWithAll(test.url))
+				if len(matches) <= 1:
+					continue
+				ruleConflictProblemsExist = True
+				logging.error("%s: Test url '%s' can be rewritten in several ways:\n%s" % (
+					ruleset.filename,
+					test.url,
+					'\n'.join(imap(
+						lambda match: '  %s: %s' % (match.ruleset.filename, match.url),
+						matches
+					))
+				))
```

Comment:
Would you mind unnesting this a bit to make it easier to read?
```python
                                msg = "%s: Test url '%s' can be rewritten in several ways:\n%s"
				ways = '\n'.join(
					imap(lambda match: '  %s: %s' % (match.ruleset.filename, match.url),
					     matches))
				logging.error(msg % (ruleset.filename, test.url, ways))
```
Something like this would work, or with hanging closing brackets.